### PR TITLE
Enable commercial solver use without modifying GenX source code

### PR DESCRIPTION
**Pull request by gmantegna**
## Description

This PR enables users to use a commercial solver without modifying the GenX source code, by adding an optimizer object as an argument to run_genx_case. This will both eliminate the significant annoyance for developers who use Gurobi of needing to update the GenX environment and module every time they pull code changes, and make it possible for users that are simply installing GenX as a package (without the ability to modify the source code) to use solvers that are not HiGHS. It also conforms with best practices suggested by the developers of JuMP (see https://github.com/jump-dev/Gurobi.jl/issues/517).

A couple of notes:
- Despite the new argument to run_genx_case, backward compatibility is maintained when using the HiGHS solver, as the optimizer object is an optional argument with HiGHS as a default.
- Users using non-HiGHS solvers will need to update their Run.jl files with this code change. However, they have to update the GenX source code anyways when pulling new changes so I do not anticipate this being a barrier.
- The type for the optimizer object is currently set to Any, which JuMP developers have confirmed is the proper type to set.

## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Code Refactor
- [ ] Performance Improvements

## Related Tickets & Documents

https://github.com/jump-dev/Gurobi.jl/issues/517

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and .md files under /docs/src have been updated if necessary.
- [] The latest changes on the target branch have been incorporated, so that any conflicts are taken care of before merging. This can be accomplished either by merging in the target branch (e.g. 'git merge develop') or by rebasing on top of the target branch (e.g. 'git rebase develop'). Please do not hesitate to reach out to the GenX development team if you need help with this.
- [x] Code has been tested to ensure all functionality works as intended.
- [x] CHANGELOG.md has been updated (if this is a 'notable' change).
- [x] I consent to the release of this PR's code under the GNU General Public license.

## How this can be tested

Can test using the SmallNewEngland/OneZone_Gurobi test case.

## Post-approval checklist for GenX core developers
After the PR is approved

- [x] Check that the latest changes on the target branch are incorporated, either via merge or rebase
- [x] Remember to squash and merge if incorporating into develop

### CODE REVIEW

File: src/GenX.jl
Comment (gmantegna) [reply src/GenX.jl:35]: Thanks. @odow do you have any idea how to properly specify the type for an Optimizer object?

File: src/case_runners/case_runner.jl
Comment (odow) [inline src/case_runners/case_runner.jl:16]: `::Any` here is correct, because users can pass one of the following:

 * a `Type{<:MOI.AbstractOptimizer}`, like `HiGHS.Optimizer`
 * a function which when called with zero arguments returns an optimizer, like `() -> Gurobi.Optimizer(env)`

**Comment by gmantegna** [comment]
Awesome thanks so much.

**Comment by cfe316** [comment]
Here were some comments from the group discussion today. I think the sentiment was positive.

* Please update the documentation to reflect how to use this.
  * Emphasize that Highs is the default 
* Add to the changelog.
* Make the example simpler: currently it's an 8760 example; make it an even-more-minimal commerical-solver example. Perhaps 24 hours, as few generators and policies as possible etc.

**Review by cfe316 — COMMENTED** [review]
This will be really excellent, and a key step for making GenX into a proper package.

* My main recommendation is to chop down the example to something like 24 hours.
* Remove all solver settings other than highs/gurobi/cplex
* suggest calling the example ProprietarySolver or CommercialSolver rather than OneZone_Gurobi
* alter its README to stress what this example is about; forget the generic SmallNewEngland instructions

**Review by cfe316 — CHANGES_REQUESTED** [review]
Please see requested changes about - then let's pull it in!

**Comment by cfe316** [comment]
@lbonaldo This PR needs to be rebased on the latest develop.
`Load_data.csv` in the example will need to be changed to `Demand_data.csv`.

**Review by cfe316 — APPROVED** [review]
I believe all changes are in order. @lbonaldo any reason we can't merge this?

**Comment by lbonaldo** [comment]
No, everything looks ready. We will have a final review tomorrow and then merge it.

**Pull request merged**
